src/routes/manage_job/get.rs

```diff
@@ -47,6 +73,338 @@ pub struct JobStatusResponseBody {
+							"is_valid_syntax" => {
+								syntax_is_valid_syntax =
+									val.as_bool().ok_or("is_valid_syntax should be a boolean")?
+							}
+							"username" => {
+								syntax_username = val
+									.as_str()
+									.ok_or("username should be a string")?
+									.to_string()
+							}
+							"domain" => {
+								syntax_domain =
+									val.as_str().ok_or("domain should be a string")?.to_string()
+							}
+							_ => {}
+						}
+					}
+				}
+				// ignore unknown fields
+				_ => {}
+			}
+		}
+
+		Ok(JobResultCsvResponse {
+			input,
+			is_reachable,
+			misc_is_disposable,
+			misc_is_role_account,
+			mx_accepts_mail,
+			smtp_can_connect,
+			smtp_has_full_inbox,
+			smtp_is_catch_all,
+			smtp_is_deliverable,
+			smtp_is_disabled,
+			syntax_domain,
+			syntax_is_valid_syntax,
+			syntax_username,
+			error,
+		})
+	}
+}
+
+async fn job_result(
+	job_id: i32,
+	req: JobResultRequest,
+	conn_pool: Pool<Postgres>,
+	// ) -> Either<Result<impl warp::Reply, warp::Rejection>, Result<impl warp::Reply, warp::Rejection>> {
+) -> Result<impl warp::Reply, warp::Rejection> {
+	match req.format {
+		JobResultResponseFormat::Json => {
+			let data = job_result_json(
+				job_id,
+				req.limit.unwrap_or(50),
+				req.offset.unwrap_or(0),
+				conn_pool,
+			)
+			.await?;
+
+			let reply =
+				serde_json::to_vec(&JobResultJsonResponse { results: data }).map_err(|e| {
+					log::error!(
+						target:"reacher",
+						"Failed to convert json results to string for [job_id={}] with [error={}]",
+						job_id,
+						e
+					);
+
+					ReacherError::Json()
+				})?;
+
+			Ok(warp::reply::with_header(
+				reply,
+				"Content-Type",
+				"application/json",
+			))
+		}
+		JobResultResponseFormat::Csv => {
+			let data = job_result_csv(
+				job_id,
+				req.limit.unwrap_or(5000),
+				req.offset.unwrap_or(0),
+				conn_pool,
+			)
+			.await?;
+
+			Ok(warp::reply::with_header(data, "Content-Type", "text/csv"))
+		}
+	}
+}
+
+async fn job_result_csv(
+	job_id: i32,
+	limit: u64,
+	offset: u64,
+	conn_pool: Pool<Postgres>,
+) -> Result<Vec<u8>, warp::Rejection> {
+	let query = sqlx::query!(
+		r#"
+		SELECT result FROM email_results
+		WHERE job_id = $1
+		ORDER BY id
+		LIMIT $2 OFFSET $3
+		"#,
+		job_id,
+		limit as i64,
+		offset as i64
+	);
+
+	let mut wtr = WriterBuilder::new().has_headers(true).from_writer(vec![]);
+
+	for json_value in conn_pool
+		.fetch_all(query)
+		.await
+		.map_err(|e| {
+			log::error!(
+				target:"reacher",
+				"Failed to get results for [job_id={}] with [error={}]",
+				job_id,
+				e
+			);
+
+			ReacherError::from(e)
+		})?
+		.iter()
+		.map(|row| row.get("result"))
+	{
+		let result_csv: JobResultCsvResponse = CsvWrapper(json_value).try_into().map_err(|e| {
+			log::error!(
+				target:"reacher",
+				"Failed to convert json to csv output struct for [job_id={}] [limit={}] [offset={}] to csv with [error={}]",
+				job_id,
+				limit,
+				offset,
+				e
+			);
+
+			ReacherError::Csv()
```

Comment:
Why don't we want to embed the `e` error inside CSV? Is is because of serialization?

Reply:
I think it's fine as-is